src/plugins/data/common/index_patterns/fields/field_list.ts

```diff
@@ -80,6 +82,12 @@ export class FieldList extends Array<IndexPatternField> implements IIndexPattern
     ...(this.groups.get(type) || new Map()).values(),
   ];
   public readonly add = (field: FieldSpec) => {
+    const displayNameMap = this.indexPattern.attributes?.fields;
+    field.displayName =
+      displayNameMap && displayNameMap[field.name] && displayNameMap[field.name].displayName
+        ? displayNameMap[field.name].displayName
+        : '';
+
     const newField = new IndexPatternField(
```

Comment:
_Instead of looking up the name, pass it to the IndexPatternField instance_
this is what I've changed, wonder if it is the way you meant

Reply:
I'm working on some changes made necessary by the removal of the index patterns reference in the field and field list in this pr - https://github.com/elastic/kibana/pull/75185